Doc/c-api/long.rst

```diff
@@ -529,10 +529,190 @@ distinguished from a number.  Use :c:func:`PyErr_Occurred` to disambiguate.
+
+   See :c:member:`PyLong_LAYOUT.bits_per_digit` for the number of bits per
+   digit and :c:member:`PyLong_LAYOUT.digit_size` for the size of a digit (in
+   bytes).
+
+
+.. c:struct:: PyLong_LAYOUT
+
+   Layout of an array of digits, used by Python :class:`int` object.
+
+   See also :attr:`sys.int_info` which exposes similar information to Python.
+
+   .. c:member:: uint8_t bits_per_digit
+
+      Bits per digit.
+
+   .. c:member:: uint8_t digit_size
+
+      Digit size in bytes.
+
+   .. c:member:: int8_t word_endian
+
+      Word endian:
+
+      - ``1`` for most significant byte first (big endian)
+      - ``-1`` for least significant first (little endian)
+
+   .. c:member:: int8_t array_endian
+
+      Array endian:
+
+      - ``1`` for most significant byte first (big endian)
+      - ``-1`` for least significant first (little endian)
+
+
+.. c:function:: PyObject* PyLong_Import(int negative, size_t ndigits, Py_digit *digits)
+
+   Create a Python :class:`int` object from an array of digits.
+
+   On success, return a Python :class:`int` object.
+   On error, set an exception and return ``NULL``.
+
+   *negative* is ``1`` if the number is negative, or ``0`` otherwise.
+
+   *ndigits* is the number of digits in the *digits* array.
+
+   *digits* is an array of unsigned digits. Digits must be in the range
+   [``0``; ``PyLong_BASE - 1``].
+
+   See :c:struct:`PyLong_LAYOUT` for the layout of an array of digits.
+
+   See also the :ref:`PyLongWriter API <pylongwriter>`.
+
+
+.. c:struct:: PyLong_DigitArray
+
+   A Python :class:`int` object exported as an array of digits.
+
+   See :c:struct:`PyLong_LAYOUT` for the :c:member:`digits` layout.
+
+   .. c:member:: PyLongObject *obj
+
+      Strong reference to the Python :class:`int` object.
+
+   .. c:member:: int negative
+
+      1 if the number is negative, 0 otherwise.
+
+   .. c:member:: size_t ndigits
+
+      Number of digits in :c:member:`digits` array.
+
+   .. c:member:: Py_digit *digits
+
+      Array of unsigned digits.
+
+
+.. c:function:: int PyLong_Export(PyObject *obj, PyLong_DigitArray *array)
+
+   Export a Python :class:`int` object as an array of digits.
+
+   On success, set *\*array* and return 0.
+   On error, set an exception and return -1.
+
+   This function always succeeds if *obj* is a Python :class:`int` object or a
+   subclass.
+
+   :c:func:`PyLong_ReleaseExport` must be called once done with using
+   *export*.
+
+
+.. c:function:: void PyLong_ReleaseExport(PyLong_DigitArray *array)
+
+   Release the export *array* created by :c:func:`PyLong_Export`.
+
+
+.. _pylongwriter:
+
+PyLongWriter API
+^^^^^^^^^^^^^^^^
+
+The :c:func:`PyLong_Import` function can be used to create a Python
+:class:`int` from an array of digits. When converting an integer from a format
+to a Python :class:`int`, creating a temporary array of digits can be
+inefficient. In this case, the :c:type:`PyLongWriter` API can be used for even
+more efficient import.
```

Comment:
I doubt this will help users.  I.e. it's not clear when PyLong_Import() may be inefficient and we can't provide this information.

Lets keep one way to do things.  PyLong_Import() is a redundant function, it should be removed; and something like PyLongWriter_Create/Finish - simple enough interface for writing.

Reply:
Ok, I removed PyLong_Import().